### PR TITLE
UX-93 Storybook welcome page

### PR DESCRIPTION
**Pull request by phoenixeliot**
Initial draft, needs to be hooked in to actually display as the welcome page (probably as an MDX story?)

### CODE REVIEW

File: stories/Welcome-draft.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+```jsx
+import { CheckCircleOutlined } from '@marklogic/design-system/src/MLIcon'
+```
+
+
+## Storybook Gotchas
+
+How StoryBook renders Story Source blocks:
+- Source blocks show what React components are rendered, and their props (and any other HTML elements in the story). Other source code that goes into a story does not generally display in the Story Source panel.
+- When showing you the source code for a story, StoryBook displays function props as `noRefCheck`, rather than showing the function contents. When copying from these stories, you will have to provide actual functions here instead. In most cases, Ant is better reference for what functions are appropriate for these props.
```
Comment (DanLipsitt) [inline stories/Welcome-draft.md]: This is fixed in the new DocsPage Storybook addon.

File: stories/Welcome-draft.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+Some components are nested inside others, eg `MLInput.MLGroup`:
+- These generally are matched to how Ant organizes their components
+- Like Ant, these have displayNames like `MLInputGroup` in Story Source panels, but should be used like so:
+```jsx
+import { MLInput } from '@marklogic/design-system'
+const { MLGroup } = MLInput // or use MLInput.MLGroup directly, in your JSX
+```
+
+A few components must be imported directly from their path, or should be if you want tree-shaking (like with {linkTo('MLIcon')}):
```
Comment (DanLipsitt) [inline stories/Welcome-draft.md]: How do we tell which components we need to do this with?

File: stories/Welcome-draft.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+
+## How to customize theme colors
+
+In your webpack.config.js or config-overrides.js, provide the Less loader the `modifyVars` option, defining any `less` variables you want to override. There is an example at `@marklogic/design-system/src/theme-variables.json`.
+
+See [Ant's page on customizing the theme](https://ant.design/docs/react/customize-theme) for more detail and a full list of theme variables provided by Ant.
+
+
+## How to use these components
```
Comment (DanLipsitt) [inline stories/Welcome-draft.md]: I assume we don't need install info if people are getting the library from npm?

File: stories/Welcome-draft.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import { linkTo } from '@storybook/addon-links';
+
+TODO: Display this as a proper MDX file.
+
+# Welcome to the MarkLogic Design System Storybook
```
Comment (DanLipsitt) [inline stories/Welcome-draft.md]: It might make sense to have a brief overview of what this storybook contains and what it is for.

**Review by DanLipsitt — CHANGES_REQUESTED** [review]

File: stories/Welcome-draft.md (hunk 1 above)
Comment (phoenixeliot) [reply stories/Welcome-draft.md]: I'll update this text when that gets merged, then

File: stories/Welcome-draft.md (hunk 3 above)
Comment (phoenixeliot) [reply stories/Welcome-draft.md]: I'm confused what you mean here, and which part of the code above you're referring to

File: stories/Welcome-draft.md (hunk 2 above)
Comment (phoenixeliot) [reply stories/Welcome-draft.md]: I've updated it to just be about icons; the others are ones you don't need to be using directly anyway.

File: stories/Welcome-draft.md (hunk 3 above)
Comment (DanLipsitt) [reply stories/Welcome-draft.md]: I thought I deleted this comment. Nothing to see here... 😊

Comment (DanLipsitt) [reply stories/Welcome-draft.md]: (ML said in our last call that we can assume that library install is documented elsewhere.)

**Review by DanLipsitt — DISMISSED** [review]
All good now. (Pending integration to display properly in the storybook.)

**Comment by DanLipsitt** [comment]
@phoenixeliot Possibly another quirk for the gotchas list: icons will look different in the storybook code view depending on which icon set they’re from (I think?). For example, `<MLIcon.Route />` gets rendered as `<route />` and <MLIcon.SearchOutlined /> gets rendered as `<SearchOutlined />` . Things will work fine if they follow your instructions, but I’m not sure what will happen if they try to paste `<route />` , so you may want a specific warning about that. It might also be worth noting that following the Ant Icon docs may not work.

**Review by brucean52 — DISMISSED** [review]

**Comment by phoenixeliot** [comment]
I updated the text to reflect how FontAwesome icons should be accessed. I fixed the icon displayName capitalization in another branch (26_result), so I've moved that small fix here so the documentation is at least accompanied by the thing that makes it correct.

**Review by brucean52 — APPROVED** [review]

**Pull request merged**